src/vs/platform/dialogs/electron-main/dialogMainService.ts

```diff
@@ -157,21 +172,31 @@ export class DialogMainService implements IDialogMainService {
 			return path;
 		}
 
-		return this.getDialogQueue(window).queue(async () => {
-			let result: SaveDialogReturnValue;
-			if (window) {
-				result = await dialog.showSaveDialog(window, options);
-			} else {
-				result = await dialog.showSaveDialog(options);
-			}
+		// prevent duplicates of the same dialog queueing at the same time
+		const fileDialogLock = await this.acquireFileDialogLock(options, window);
```

Comment:
Why is `acquireFileDialogLock` awaited, this method is not using any promises? I would remove the `async` from it.

Reply:
Oops, fixed in 1e9e036